hedera-node/src/main/java/com/hedera/services/sigs/order/SigRequirements.java

```diff
@@ -1113,8 +1113,10 @@ private KeyOrderingFailure includeIfNecessary(
 		if (!payer.equals(account)) {
 			var result = sigMetaLookup.aliasableAccountSigningMetaFor(account, linkedRefs);
 			if (result.succeeded()) {
-				var meta = result.metadata();
-				if (adjust.getAmount() < 0 || meta.receiverSigRequired()) {
+				final var meta = result.metadata();
+
+				if (((adjust.getAmount() < 0  && !adjust.getIsApproval()) || meta.receiverSigRequired())) {
```

Comment:
Imo it's easier to read conditions that are limited to at most two factors; so instead e.g.,
```
final var isUnapprovedDebit = adjust.getAmount() < 0  && !adjust.getIsApproval();
if (isUnapprovedDebit || meta.receiverSigRequired()) {
  ...
```